src/components/comment-box.jsx

```diff
@@ -1,22 +1,74 @@
 import React, { PropTypes } from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
 
+import * as commentActions from '../actions/commentActions';
 import CommentForm from './comment-form';
 import CommentList from './comment-list';
 
-const CommentBox = (props) => (
-  <div className="commentBox">
-    <h1>Comments</h1>
-    <CommentList comments={props.comments} />
-    <CommentForm handleSubmit={props.handleSubmit} />
-  </div>
-);
-
-CommentBox.propTypes = {
-  comments: PropTypes.arrayOf(PropTypes.shape({
-    author: PropTypes.string,
-    message: PropTypes.string,
-  })),
-  handleSubmit: PropTypes.func.isRequired,
-};
-
-export default CommentBox;
+class CommentBox extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      comment: Object.assign({}, this.props.comment),
+      isSaving: false
+    };
+
+    this.updateComment = this.updateComment.bind(this);
+    this.saveComment = this.saveComment.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.actions.loadComments();
+  }
+
+  updateComment(event) {
+    const field = event.target.name;
+    let comment = this.state.comment;
+    comment[field] = event.target.value;
+    this.setState({comment: comment});
+  }
+
+  saveComment(event) {
+    event.preventDefault();
+    this.setState({isSaving: true});
+
+    this.props.actions.saveComment(this.state.comment).then(() => {
+      this.setState({isSaving: false});
+    });
+  }
+
+  render() {
+    const {comments, isLoading} = this.props;
+
+    return (
+      <div className="commentBox">
+        <h1>Comments</h1>
+        <CommentList comments={comments} isLoading={isLoading} />
+        <CommentForm
+          comment={this.state.comment}
+          onChange={this.updateComment}
+          onSave={this.saveComment}
+          isSaving={this.state.isSaving} />
```

Comment:
I thought this could be a good place to go over smart/dumb, presentaion/container components. 

I'm treating `CommentForm` as a purely presentational component. The API call to save a `comment` is done here in `CommentBox` the form's container. The container component knows when a `save` is complete that's why I'm passing it as a `prop` to the form. 

What are you thinking of doing?